src/Microsoft.DotNet.Tools.Pack/InputOutputPathsCalculator.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Tools.Pack
+{
+    public class InputOutputPathsCalculator
+    {
+        private readonly Project _project;
+
+        private readonly string _configuration;
+
+        private bool OutputPathSet => !string.IsNullOrWhiteSpace(OutputPathParameter);
+
+        public string InputPathParameter { get; }
+
+        public string OutputPathParameter { get; }
+
+        public bool InputPathSet => !string.IsNullOrWhiteSpace(InputPathParameter);        
+
+        public string InputPath => InputPathSet ? InputPathParameter : OutputPath;
+
+        public string OutputPath
+        {
+            get
+            {
+                if (OutputPathSet)
+                {
+                    return OutputPathParameter;
+                }
+
+                var outputPath = Path.Combine(
+                    _project.ProjectDirectory,
+                    Constants.BinDirectoryName,
+                    _configuration);
+
+                return outputPath;
+            }
+        }        
+
+        public InputOutputPathsCalculator(Project project, string inputPath, string outputPath, string configuration)
```

Comment:
What is this extension method? What does it do? 

I just moved code around from Program.cs to here. What this class is doing is pretty much what was being done in program.cs plus figuring out the right InputPath, which will depend on wether an input option is passed (we use it), or not (in which case we use OutputPath).